File: app/src/main/java/com/arnaudpiroelle/manga/provider/japscan/api/JapScanApiService.java
package com.arnaudpiroelle.manga.provider.japscan.api;

import retrofit.client.Response;
import retrofit.http.GET;
import retrofit.http.Path;

public interface JapScanApiService {
    @GET("/mangas/")
    Response getMangaList();

    @GET("/mangas/{mangaAlias}/")
    Response getManga(@Path("mangaAlias") String mangaAlias);

    @GET("/lecture-en-ligne/{mangaAlias}/{chapterNumber}")
    Response getReader(@Path("mangaAlias") String mangaAlias, @Path("chapterNumber") String chapterNumber);

    @GET("/lecture-en-ligne/{mangaAlias}/{chapterNumber}/{pageNumber}.html")
    Response getPage(@Path("mangaAlias") String mangaAlias, @Path("chapterNumber") String chapterNumber, @Path("pageNumber") String pageNumber);
}

File: app/src/main/kotlin/com/arnaudpiroelle/manga/provider/japscan/api/JapScanApiService.kt
package com.arnaudpiroelle.manga.provider.japscan.api

import retrofit.client.Response
import retrofit.http.GET
import retrofit.http.Path

interface JapScanApiService {
    @GET("/mangas/")
    fun getMangaList(): Response

    @GET("/mangas/{mangaAlias}/")
    fun getManga(@Path("mangaAlias") mangaAlias: String): Response

    @GET("/lecture-en-ligne/{mangaAlias}/{chapterNumber}")
    fun getReader(@Path("mangaAlias") mangaAlias: String, @Path("chapterNumber") chapterNumber: String): Response

    @GET("/lecture-en-ligne/{mangaAlias}/{chapterNumber}/{pageNumber}.html")
    fun getPage(@Path("mangaAlias") mangaAlias: String, @Path("chapterNumber") chapterNumber: String, @Path("pageNumber") pageNumber: String): Response
}
